tests/unit/s2n_key_update_test.c

```diff
@@ -148,34 +150,36 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
         };
 
-        /* Key update messages not allowed with ktls */
+        /* Key update messages not allowed with ktls unless the platform supports it */
         {
             DEFER_CLEANUP(struct s2n_stuffer input, s2n_stuffer_free);
             EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, 0));
 
             DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
                     s2n_connection_ptr_free);
             EXPECT_NOT_NULL(conn);
-            conn->actual_protocol_version = S2N_TLS13;
-            EXPECT_NOT_NULL(conn->secure);
-            conn->secure->cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
+            EXPECT_OK(s2n_test_configure_connection_for_ktls(conn, S2N_TLS13, &s2n_tls13_aes_128_gcm_sha256));
 
-            /* Fails if receiving with ktls:
-             * Kernel key update would be required.
-             */
+            /* Only able to read ktls key updates if platform supports it */
             conn->ktls_send_enabled = true;
             conn->ktls_recv_enabled = true;
             EXPECT_SUCCESS(s2n_stuffer_write_uint8(&input, S2N_KEY_UPDATE_NOT_REQUESTED));
-            EXPECT_FAILURE_WITH_ERRNO(s2n_key_update_recv(conn, &input), S2N_ERR_KTLS_KEYUPDATE);
+            if (s2n_ktls_keyupdate_is_supported_on_platform()) {
+                EXPECT_SUCCESS(s2n_key_update_recv(conn, &input));
+            } else {
+                EXPECT_FAILURE_WITH_ERRNO(s2n_key_update_recv(conn, &input), S2N_ERR_KTLS_KEYUPDATE);
+            }
```

Comment:
This works pretty well while there exists platforms without keyupdates, bc the S2N_ERR_KTLS_KEYUPDATE error proves that the key update is (not) happening. But I'm a little worried about the future where we never take the S2N_ERR_KTLS_KEYUPDATE branch. How would we know this test is still working and the keyupdate is actually happening?

This might hold for the other tests too. I think we might need more recv/send_key_updated assertions.